backend/molgenis-emx2-graphql/src/test/java/org/molgenis/emx2/graphql/TestGraphqlPermissions.java

```diff
@@ -0,0 +1,403 @@
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.StreamSupport;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.molgenis.emx2.Database;
+import org.molgenis.emx2.MolgenisException;
+import org.molgenis.emx2.Schema;
+import org.molgenis.emx2.sql.TestDatabaseFactory;
+import org.molgenis.emx2.tasks.TaskService;
+import org.molgenis.emx2.tasks.TaskServiceInMemory;
+
+@TestMethodOrder(OrderAnnotation.class)
+public class TestGraphqlPermissions {
+
+  private static GraphQL graphQLSchema;
+  private static GraphQL graphQLDatabase;
+  private static Database database;
+  private static GraphqlSessionHandlerInterface sessionManager;
+  private static final String SCHEMA_NAME = "TestGraphqlPermissions";
+
+  // Carries row IDs across ordered tests that depend on prior state.
+  private static String orderIdOwnedBySpecial;
+  private static String orderIdOwnedByEditor;
+
+  @BeforeAll
+  static void setup() {
+    database = TestDatabaseFactory.getTestDatabase();
+    TaskService taskService = new TaskServiceInMemory();
+    Schema schema = database.dropCreateSchema(SCHEMA_NAME);
+    PET_STORE.getImportTask(schema, true).run();
+    graphQLDatabase = new GraphqlApiFactory().createGraphqlForDatabase(database, taskService);
+    graphQLSchema = new GraphqlApiFactory().createGraphqlForSchema(schema);
+
+    sessionManager =
+        new GraphqlSessionHandlerInterface() {
+          private String user;
+
+          @Override
+          public void createSession(String username) {
+            this.user = username;
+          }
+
+          @Override
+          public void destroySession() {
+            this.user = null;
+          }
+
+          @Override
+          public String getCurrentUser() {
+            return user;
+          }
+        };
+
+    if (database.hasUser("testViewer")) database.removeUser("testViewer");
+    if (database.hasUser("testEditorSpecial")) database.removeUser("testEditorSpecial");
+    if (database.hasUser("testEditor")) database.removeUser("testEditor");
+  }
+
+  @Test
+  @Order(1)
+  void createPermissionGroups() throws IOException {
+    executeDb("mutation{signup(email:\"testViewer\",password:\"test123456\"){message}}");
+    executeDb("mutation{signup(email:\"testEditorSpecial\",password:\"test123456\"){message}}");
+    executeDb("mutation{signup(email:\"testEditor\",password:\"test123456\"){message}}");
+    executeSchema(
+        """
+        mutation {
+          change(permissions: [
+            {
+              groupName: "TestGraphqlPermissions/editorSpecial",
+              tableId: "Order",
+              isRowLevel: true,
+              hasSelect: true, hasInsert: true, hasUpdate:true, hasDelete:true, hasAdmin:false,
+              users: ["testEditorSpecial", "someOtherUser@test.com"]
```

Comment:
The relation is:

USER  0...n <- GROUP -> 0..1 PERMISSIONS

So I don't understand why we can combine this call like this.

Reply:
If you would like one mutation format that covers it all (which is not too bad an idea) then expected more something like

{group:'my group', permissions: [{schema:'x', table: '_all',  hasSelect: true, hasInsert: true, hasUpdate:true, hasDelete:true, hasAdmin:false},{schema:'x', table: 'z',  hasSelect: true, hasInsert: true, hasUpdate:true, hasDelete:true, hasAdmin:false}], users: ["testEditorSpecial", "someOtherUser@test.com"]}